openedx/course-discovery
# feat: add watchers to the courses

[PROD-3412](https://2u-internal.atlassian.net/browse/PROD-3412)
----------------------
This PR adds the watchers field to `Course` model and adds the `watchers` field to the `CourseSerializer` to add the watchers field to the API response.
This PR also enables email notification for the `watchers` of the course when the `Course Run` of the course is published or published.


Testing:
----------------------
- Run `course-discovery`, `frontend-app-publisher`, `studio` and `ecommerce` locally.
- Activate` discovery` shell `make shell-discovery`
- Run make requirements to install the requirements for `django-multi-email-field`
- Run `./manage.py migrate course_metadata` to migrate the changes.
- Add watchers of the course in the `Course` model through the Django admin.
- Check if wachers field is present in the API response of the course.
- To check the email notification, you can clone this frontend PR (https://github.com/openedx/frontend-app-publisher/pull/889) (Currently WIP only need to replace text field with multi email field).
- Fill watchers field with the email comma separated email ids.
- Publish & Schedule the course run of the course through the publisher. Add the `pdb.set_trace()` in the `send_email_to_watchers` to check if the email method is called. Because on local it will produce error due to missing email credentials.

### Screenshots:
Email Template: 
(Write HTML file on local as it will give credential error in sending email function)
<img width="1680" alt="image" src="https://github.com/openedx/course-discovery/assets/78806673/b2814f84-c917-4841-b934-ae8c732a29a4">
